internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -264,11 +299,56 @@ func (c *FunctionComposer) Compose(ctx context.Context, xr *composite.Unstructur
 			req.Input = in
 		}
 
-		// TODO(negz): Generate a content-addressable tag for this request.
-		// Perhaps using https://github.com/cerbos/protoc-gen-go-hashpb ?
-		rsp, err := c.pipeline.RunFunction(ctx, fn.FunctionRef.Name, req)
-		if err != nil {
-			return CompositionResult{}, errors.Wrapf(err, errFmtRunPipelineStep, fn.Step)
+		// used to store the requirements returned at the previous iteration
+		var requirements *v1beta1.Requirements
+		// used to store the response of the function at the previous iteration
+		var rsp *v1beta1.RunFunctionResponse
+
+		for i := int64(0); i <= MaxRequirementsIterations; i++ {
+			// if we reached the maximum number of iterations we need to return an error
```

Comment:
I'd maybe mention requirement stabilization in this comment too, otherwise its not saying much that the code doesn't already document pretty clearly.